src/coreclr/vm/jitinterface.cpp

```diff
@@ -11822,6 +11873,74 @@ void* CEEJitInfo::getFieldAddress(CORINFO_FIELD_HANDLE fieldHnd,
     return result;
 }
 
+bool CEEInfo::getReadonlyStaticFieldValue(CORINFO_FIELD_HANDLE fieldHnd, uint8_t* buffer, int bufferSize)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    _ASSERT(fieldHnd != NULL);
+    _ASSERT(buffer != NULL);
+    _ASSERT(bufferSize > 0);
+
+    bool result = false;
+
+    JIT_TO_EE_TRANSITION();
+
+    FieldDesc* field = (FieldDesc*)fieldHnd;
+    if (field->IsStatic() && !field->IsThreadStatic() && ((unsigned)bufferSize == field->GetSize()))
```

Comment:
Note that `FieldDesc::GetSize` is moderately expensive for structs (it hits typeloader). No change needed ... just pointing it out.

Reply:
Ah, good to know, thanks

Reply:
Thinking about this more. This whole condition can be asserts - it is duplicating checks done by the JIT.